### DISCUSSION TITLE
Converting RGB data to XYZ to plot changes in color over time

### DISCUSSION THREAD

**Discussion by LandonSwartz** [colour-science/colour#1027] [Q&A]
Hello, 
I'm working on a project collecting images of plants over time using robotics. Through the images collected, we are able to find RGB color values of the plants. We are wanting to illustrate changes in color over time for a normal plant and stressed plant (one stays green, the other turns yellower over time). We thought of using CIE color graphs to show the difference but I was wanting to ask for a community opinion on the best plotting method for illustrating changes in color. Furthermore, do you need anything special (i.e. illuminati) to convert from RGB to XYZ for plotting? I appreciate any help I can get. 

[example_data_set.ods](https://github.com/colour-science/colour/files/9285428/example_data_set.ods)

**Comment by thomasmansencal** [comment]
Hi @LandonSwartz,

If you know which RGB colourspace your data represent, it should be trivial to plot it on a chromaticity diagram. Is there any particular reason why you would want to use XYZ?

Cheers,

Thomas

**Reply by LandonSwartz** [reply]
I was converting RGB to XYZ to XY to plot on a chromaticity diagram. The problem is the resolution of telling differences isn't super amazing because I think I am doing something wrong. Attached is a photo showing some diagrams I was able to get:
![plot_rgb_cie_plot](https://user-images.githubusercontent.com/50836209/183688614-e9420bea-0053-4efa-8849-c69f4c72d41f.png)

Would a color difference formula be helpful? Also how do I make sure I know my RGB colorspace? My camera is this: https://www.alliedvision.com/en/camera-selector/detail/mako/G-503/. Do I just assume a normal Adobe 1998 RGB setting or something else?

Thank you for the reply also! I appreciate the help.
Landon

**Reply by LandonSwartz** [reply]
Furthermore, what would be a good method to plot color changes throughout time? I've done multi color swatches but it doesn't show the CIE colors. I am very unfamiliar with color science but studied CS so I am trying to figure it out mathematically.

**Reply by thomasmansencal** [reply]
You could maybe zoom in the chromaticity diagram?
